lms/templates/edxnotes/edxnotes.html

```diff
@@ -112,7 +111,7 @@ <h2 class="placeholder-title">${_('You have not made any notes in this course ye
     <%static:require_module module_name="js/edxnotes/views/page_factory" class_name="NotesPageFactory">
         NotesPageFactory({
             disabledTabs: ${disabled_tabs},
-            notes: ${notes},
+            notes: ${dump_js_escaped_json(notes, NoteJSONEncoder) | n},
```

Comment:
@muzaffaryousaf Our current best practice recommends that everything gets escaped.  That way we don't need to think about what may or may not contain user data to determine what templates are safe.

For strings, this is simple:
```
    notesEndpoint: '${notes_endpoint | n, js_escaped_string}',
```
For numbers like page_size this would be:
```
     pageSize: '${page_size | n, dump_js_escaped_json}',
```
For other data, it seems it is already JSON?  Like debugMode is "true" or "false"?  If so, this would be done as follows:
```
    debugMode: ${json.loads(debug) | n, dump_js_escaped_json}
```

However, the recommendation is also that you should avoid this, and everything should by Python until it needs to be something else.  So, if debug were instead True or False, you would only need to have the following:
```
    debugMode: ${debug | n, dump_js_escaped_json}
```

The point of the best practice is that all expression in the platform will eventually be escaped and it is less likely that we have any vulnerabilities.